File: actin-common/src/main/java/com/hartwig/actin/clinical/datamodel/PriorOtherCondition.java
package com.hartwig.actin.clinical.datamodel;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class PriorOtherCondition {

    @NotNull
    public abstract String name();

    @NotNull
    public abstract String nameDoid();

    @NotNull
    public abstract String category();

    @NotNull
    public abstract String categoryDoid();

}

File: datamodel/src/main/kotlin/com/hartwig/actin/datamodel/clinical/PriorOtherCondition.kt
package com.hartwig.actin.datamodel.clinical

data class PriorOtherCondition(
    override val name: String,
    override val year: Int? = null,
    override val month: Int? = null,
    override val icdCodes: Set<IcdCode>,
    val isContraindicationForTherapy: Boolean,
): Comorbidity {
    override val comorbidityClass = ComorbidityClass.PRIOR_OTHER_CONDITION
}